pkg/splunk/enterprise/clustermanager.go

```diff
@@ -50,6 +50,7 @@ func ApplyClusterManager(ctx context.Context, client splcommon.ControllerClient,
 	reqLogger := log.FromContext(ctx)
 	scopedLog := reqLogger.WithName("ApplyClusterManager")
 	eventPublisher, _ := newK8EventPublisher(client, cr)
+	cr.Kind = "ClusterManager"
```

Comment:
Mostly because when the mock test runs , its not setting `Kind` field in the structure. just to. make sure Unit test case works fine. but this is automatically set in run time

Reply:
Nit: Can we do this in the UT code instead of here?

Reply:
Nit: Can we add this in the testing code itself? Same for all the other cases below.

Reply:
removed

Reply:
I have added this back as test are failing if I remove  this. I have created jira ticket to fix this in next release.